src/zmqpp/socket.cpp

```diff
@@ -341,7 +341,11 @@ bool socket::receive_raw(char* buffer, size_t& length, int const flags /* = NORM
 
 	if(result >= 0)
 	{
+#if __WIN32 || WIN32 || __WIN32__
+		length = min(length, zmq_msg_size(&_recv_buffer));
```

Comment:
I don't think this is the appropriate solution for the build problem. I suspect that std::min doesn't work due to the min macro defined in the Windows SDK headers. This could be solved by:
* defining NOMINMAX before including any Windows SDK header
* undefine min/max after including the Windows SDK headers
* put std::min in parentheses: `(std::min)(length, ...)`